clang/include/clang/Sema/Sema.h

```diff
@@ -12971,12 +12971,15 @@ void Sema::addIntelSYCLSingleArgFunctionAttr(Decl *D,
         return;
       }
     }
-    if (CI.getParsedKind() == ParsedAttr::AT_SYCLIntelMaxGlobalWorkDim) {
+    if (CI.getParsedKind() == ParsedAttr::AT_SYCLIntelMaxGlobalWorkDim ||
+        CI.getParsedKind() == ParsedAttr::AT_SYCLIntelNoGlobalWorkOffset) {
```

Comment:
I don't think this needs to be added here -- we only care about zero vs nonzero for this attribute.

Reply:
We need this diagnostic for the attribute when we pass negative value as template parameter. 

template <int SIZE>
class KernelFunctor {
public:
  // expected-error@+1{{'no_global_work_offset' attribute requires a non-negative integral compile time constant expression}}
  [[intel::no_global_work_offset(SIZE)]] void operator()() {}
};

int main() {
  //expected-note@+1{{in instantiation of template class 'KernelFunctor<-1>' requested here}}
  KernelFunctor<-1>();
  // no error expected
  KernelFunctor<1>();
}

Reply:
I don't see why this is any different than the question of whether to accept values other than 0 or 1. Why are negative numbers a problem? Any nonzero number should be treated as `true`, and zero should be treated as `false`.

Reply:
Done. I confused myself with the other attributes.  I considered only positive nonzero value as true ( i was wrong). I have removed this diagnostic since no need to check here. Thanks for the clarification.

Reply:
Thanks!